datasets/flwr_datasets/partitioner/continuous_partitioner_test.py

```diff
@@ -0,0 +1,188 @@
+        _, part1 = _dummy_setup(3, strictness=0.0, num_rows=30, seed=42)
+        _, part2 = _dummy_setup(3, strictness=0.0, num_rows=30, seed=43)
+        self.assertNotEqual(
+            part1.partition_id_to_indices, part2.partition_id_to_indices
+        )
+
+    def test_partitions_are_disjoint(self) -> None:
+        """Ensure no sample index is shared across partitions."""
+        _, partitioner = _dummy_setup(5, 0.8, 50)
+        all_indices = sum(partitioner.partition_id_to_indices.values(), [])
+        self.assertEqual(len(all_indices), len(set(all_indices)))
+
+    def test_partition_per_sample(self) -> None:
+        """Every row in its own partition (extreme non-iid case)."""
+        _, partitioner = _dummy_setup(10, 1.0, 10)
+        for i in range(10):
+            self.assertEqual(len(partitioner.load_partition(i)), 1)
+
+    def test_monotonic_increasing_across_partitions_when_sorted(self) -> None:
+        """With no shuffling and max strictness, partitions follow increasing order."""
+        _, partitioner = _dummy_setup(3, 1.0, 30, shuffle=False)
+        values: list[float] = []
+        for pid in range(3):
+            indices = partitioner.partition_id_to_indices[pid]
+            values.extend(partitioner.dataset["score"][i] for i in indices)
+        self.assertEqual(values, sorted(values))
+
+
+class TestContinuousPartitionerFailure(unittest.TestCase):
+    """Test ContinuousPartitioner failures by incorrect usage."""
+
+    def test_invalid_num_partitions(self) -> None:
+        """Raise ValueError for non-positive num_partitions."""
+        with self.assertRaises(ValueError):
+            _ = _dummy_setup(0, 0.5, 10)
+
+    def test_invalid_strictness(self) -> None:
+        """Raise ValueError for strictness outside [0, 1]."""
+        with self.assertRaises(ValueError):
+            _ = _dummy_setup(3, -0.1, 10)
+        with self.assertRaises(ValueError):
+            _ = _dummy_setup(3, 1.1, 10)
+
+    def test_missing_column_raises(self) -> None:
+        """Raise KeyError when partition_by column is missing."""
+        data = {
+            "features": list(range(10)),
+        }
+        dataset = Dataset.from_dict(data)
+        partitioner = ContinuousPartitioner(3, partition_by="missing", strictness=0.5)
+        partitioner.dataset = dataset
+        with self.assertRaises(KeyError):
+            _ = partitioner.load_partition(0)
+
+    def test_nan_value_in_column_raises(self) -> None:
+        """Raise ValueError when partition_by column contains NaN or None."""
+        data = {
+            "score": [1.0, 2.0, None, 4.0, 5.0],
+            "features": [0, 1, 2, 3, 4],
+        }
+        dataset = Dataset.from_dict(data)
+        partitioner = ContinuousPartitioner(2, partition_by="score", strictness=0.5)
+        partitioner.dataset = dataset
+        with self.assertRaises(ValueError):
+            _ = partitioner.load_partition(0)
+
+    def test_zero_stddev_raises(self) -> None:
```

Comment:
Based on my comment earlier in my first review, should we include a test that should pass that checks that the scenario that you test here passes when `strictness=0`.